Update GitHub Onboarding (#3962)
astronomer/docs: astro/first-dag-onboarding.md

---
sidebar_label: 'When Signing Up for Astro'
title: 'Run your first DAG on Astro'
id: 'first-dag-onboarding'
description: "Learn how to run your first Apache Airflow® DAG on Astro when you sign up for Astro."
---

Astro is the industry's leading managed service for [Apache Airflow®](https://airflow.apache.org/). To quickly learn how Astro works, follow the steps in this quickstart to create an Airflow environment and run your first DAG.

When you first sign up for Astro, you can choose to tailor your experience based on how you plan to use Astro. The onboarding process includes:

- Creating an Organization, Workspace, and Deployment
- Choosing an example Airflow project template to clone into your own sample GitHub Repository
- Connecting your Astro Deployment to your sample GitHub Repository, so you can deploy code from GitHub

> - Run your first DAG on Astro

Reviewer: ```suggestion
- Running your first DAG on Astro
```

This tutorial takes about 15 minutes. If you're new to Airflow and want a more in-depth tutorial, see [Airflow 101 Learning Path](https://academy.astronomer.io/path/airflow-101).

## Prerequisites

- An Astro account. To start an Astro trial and create your free trial account, see [Start a trial](trial.md).
- (Recommended) A [GitHub account](https://docs.github.com/en/get-started/signing-up-for-github).

:::info

> If you're on your organization's network and can't access Astro, make a request to allowlist the following domains on your network:

Reviewer: ```suggestion
If you're on your organization's network and can't access Astro, make a request to `allowlist` the following domains on your network:
```

Reviewer: maybe this should be formatted as code?

Author: I doublechecked the google style guide, which we fall back to for word use, and it says to use it as a noun vs a verb, but no code formatting! made this change here, will copy over to other.

- **Business** or **Personal** use
- Familiarity with Apache Airflow
- Use cases, including:
    - AI/ML
    - Business Operations
    - Reporting and Analytics
    - Extract, transform, and loading operations (ETL)
    - Other

## Step 2: Create your Oganization and Workspace

Enter a name for your Organization and Workspace. These can be changed later in the Astro UI.

An Astro _Organization_ is the highest level entity in Astro and represents a shared space for your company on Astro. An Organization is automatically created when you first sign up for Astronomer. At the Organization level, you can manage all of your users, Deployments, Workspaces, and clusters from a single place in the Astro UI. A _Workspace_ is a collection of Deployments that can be accessed by a specific group of users.

## Step 3: Select a template

An _Astro project_ contains the set of files necessary to run Airflow, including dedicated folders for your DAG files, plugins, and dependencies. Select a template Astro project to immediately get a demonstration of Airflow's orchestration capabilities.

> - **Generative AI**: Airflow is a common orchestration engines for AI/Machine Learning jobs, especially for retrieval-augmented generation (RAG).This [generative AI project](https://github.com/astronomer/templates/blob/main/generative-ai/README.md) shows an simple example of building vector embeddings for text and then performing a semantic search on the embeddings.

Reviewer: ```suggestion
- **Generative AI**: Airflow is a common orchestration engine for AI/Machine Learning jobs, especially for retrieval-augmented generation (RAG).This [generative AI project](https://github.com/astronomer/templates/blob/main/generative-ai/README.md) shows an simple example of building vector embeddings for text and then performing a semantic search on the embeddings.
```

> - **Generative AI**: Airflow is a common orchestration engines for AI/Machine Learning jobs, especially for retrieval-augmented generation (RAG).This [generative AI project](https://github.com/astronomer/templates/blob/main/generative-ai/README.md) shows an simple example of building vector embeddings for text and then performing a semantic search on the embeddings.

Reviewer: ```suggestion
- **Generative AI**: Airflow is a common orchestration engines for AI/Machine Learning jobs, especially for retrieval-augmented generation (RAG). This [generative AI project](https://github.com/astronomer/templates/blob/main/generative-ai/README.md) shows a simple example of building vector embeddings for text and then performing a semantic search on the embeddings.
```

- **ETL**: Use a template to make an example [ETL pipeline](https://github.com/astronomer/templates/blob/main/etl/README.md). This template shows an example pattern for defining an ETL workload using DuckDB as the data warehouse of choice.
- **Learning Airflow**: This example project is generated when you run 'astro dev init' using the Astro CLI. It shows a basic Astro project with Airflow components and DAGs.
- **None**: This option allows you to manually deploy a project after making a Deployment. See [Deploy code with GitHub](deploy-github-integration.md) to use a GitHub connection with an existing project in a repo. Or, you can choose to manually build and Deploy an example DAG by following the steps in [Run your first DAG with the Astro CLI](first-dag-cli.md) or [Run your first DAG with GitHub Actions](first-dag-github-actions.md).

## Step 4: Set up your Astro Deployment

You can now confirm your selected template and finish configuring your Deployment resources and whether you want to use a GitHub connection for code deploys.

> An Astro _Deployment_ is an instance of Apache Airflow that is powered by all core Airflow components, including a webserver, scheduler, and one or more workers. You deploy DAGs to a Deployment, and you can have one or more Deployments within a Workspace.

Reviewer: ```suggestion
An Astro _Deployment_ is an instance of Apache Airflow that is powered entirely by core Airflow components, including a webserver, scheduler, and one or more workers. You deploy DAGs to a Deployment, and you can have one or more Deployments within a Workspace.
```

- **Selected template**: When you finalize your Deployment, you can confirm or change the template option you selected in Step 3.
- **Git connection**: Enter your **GitHub Owner** and **Repository Owner** information to deploy code directly to an Astro Deployment when you merge changes to a specific branch. If you choose to skip connecting to GitHub, later, you can return and manually build and Deploy an example DAG by following the steps in [Run your first DAG with the Astro CLI](first-dag-cli.md) or [Run your first DAG with GitHub Actions](first-dag-github-actions.md).
- **Review Deployment**: Enter a **Name** for your Deployment and choose the **Provider** that you want to use and **Region** where you'd like to host your resources.

After you finish configuring your Deployment, click **Create your Project Repository and Deploy to Astro**.

## Step 5: Run your first DAG on Astro

> When you create your project and deploy it to Astro, it automatically creates a new Deployment and makes a private GitHub repository in your account with the template that you selected. Then, you can open Airflow and run your DAG.

Reviewer: ```suggestion
When you create your project and deploy it to Astro, this automatically creates a new Deployment and makes a private GitHub repository in your account with the template that you selected. Then, you can open Airflow and run your DAG.
```

- `https://updates.astronomer.io/`
- `https://install.astronomer.io/`
- `https://astro-<organization-short-name>.datakin.com/`
- `https://<organization-short-name>.astronomer.run/`
:::

## Step 1: Tailor your Astro experience

Based on how you plan to use Astro, you can find more resources for particular topics relevant to your needs.

- **Business** or **Personal** use
- Familiarity with Apache Airflow
- Use cases, including:
    - AI/ML
    - Business Operations
    - Reporting and Analytics
    - Extract, transform, and loading operations (ETL)
    - Other

## Step 2: Create your Oganization and Workspace

Enter a name for your Organization and Workspace. These can be changed later in the Astro UI.

An Astro _Organization_ is the highest level entity in Astro and represents a shared space for your company on Astro. An Organization is automatically created when you first sign up for Astronomer. At the Organization level, you can manage all of your users, Deployments, Workspaces, and clusters from a single place in the Astro UI. A _Workspace_ is a collection of Deployments that can be accessed by a specific group of users.

## Step 3: Select a template

> You can choose a template Astro project that shows a demonstration of particular scenarios. An _Astro project_ contains the set of files necessary to run Airflow, including dedicated folders for your DAG files, plugins, and dependencies.

Reviewer 2: ```suggestion
An _Astro project_ contains the set of files necessary to run Airflow, including dedicated folders for your DAG files, plugins, and dependencies. Select a template Astro project to immediately get a demonstration of Airflow's orchestration capabilities.
```

Reviewer 2: small NIT, and I think we could potentially remove or alter line 61 if we go with something like this?

> These templates include a fully functional Astro project for specific use cases, or just a general project for you to use to explore Airflow.

Author: ```suggestion
```
